Foundation/NSString.swift

```diff
@@ -1253,7 +1253,20 @@ extension NSString {
     }
     
     public convenience init(contentsOfFile path: String, encoding enc: UInt) throws {
-        NSUnimplemented()    
+        let readResult = try NSData.readBytesFromFileWithExtendedAttributes(path, options: [])
+        guard let cf = CFStringCreateWithBytes(kCFAllocatorDefault, UnsafePointer<UInt8>(readResult.bytes), readResult.length, CFStringConvertNSStringEncodingToEncoding(enc), true) else {
+            throw NSError(domain: NSCocoaErrorDomain, code: NSCocoaError.CoderReadCorruptError.rawValue, userInfo: [
+                "NSDebugDescription" : "Unable to Creates a string from a buffer containing characters in a specified encoding."
```

Comment:
Alternative error description suggestion: Unable to create a string using the specified encoding.